lib/ecto/adapters/postgres/connection.ex

```diff
@@ -40,6 +41,14 @@ if Code.ensure_loaded?(Postgrex.Connection) do
       end
     end
 
+    @doc """
+    Ensures a port is represented as an integer.  This
+    allows for configuration to grab data from the ENV without
+    having to convert it to an integer directly.
+    """
+    def normalize_port(port) when is_binary(port), do: to_int(port, Integer.parse(port))
+    def normalize_port(port), do: port
```

Comment:
I would fail here immediately with `String.to_integer/1` and a guard with `is_integer/1`. I think a stacktrace pointing here is more clear than one pointing to postgrex internals.